tools/pinning/common/export-pinned-dependencies.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# This script accepts a directory containing a pyproject.toml file configured
+# for use with poetry and generates and prints the pinned dependencies of that
+# file. Any dependencies on acme or those referencing certbot will be removed
+# from the output. The exported requirements are printed to stdout.
+set -euo pipefail
+
+if [ -z ${1+x} ]; then
+    echo "Usage:" >&2
+    echo "$0 PYPROJECT_TOML_DIRECTORY" >&2
+    exit 1
+fi
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+WORK_DIR="$1"
+
+if ! command -v poetry >/dev/null || [ $(poetry --version | grep -oE '[0-9]+\.[0-9]+' | sed 's/\.//') -lt 12 ]; then
+    echo "Please install poetry 1.2+." >&2
+    echo "You may need to recreate Certbot's virtual environment and activate it." >&2
+    exit 1
+fi
+
+# Old eggs can cause outdated dependency information to be used by poetry so we
+# delete them before generating the lock file. See
+# https://github.com/python-poetry/poetry/issues/4103 for more info.
+rm -rf ${REPO_ROOT}/*.egg-info
+
+cd "${WORK_DIR}"
+
+if [ -f poetry.lock ]; then
+    rm poetry.lock
+fi
+
+poetry lock >&2
+trap 'rm poetry.lock' EXIT
+
+# We need to remove local packages from the output.
+poetry export --without-hashes | sed '/^acme @/d; /certbot/d;'
```

Comment:
This was just copied verbatim from the old script, but the lines for local packages output by `poetry` look something like:
```
acme @ file:///Users/bmw/Development/github.com/certbot/certbot/acme; python_version >= "3.6"
certbot @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot; python_version >= "3.6"
certbot-apache @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-apache; python_version >= "3.6"
certbot-ci @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-ci; python_version >= "3.6"
certbot-compatibility-test @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-compatibility-test; python_version >= "3.6"
certbot-dns-cloudflare @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-cloudflare; python_version >= "3.6"
certbot-dns-cloudxns @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-cloudxns; python_version >= "3.6"
certbot-dns-digitalocean @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-digitalocean; python_version >= "3.6"
certbot-dns-dnsimple @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-dnsimple; python_version >= "3.6"
certbot-dns-dnsmadeeasy @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-dnsmadeeasy; python_version >= "3.6"
certbot-dns-gehirn @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-gehirn; python_version >= "3.6"
certbot-dns-google @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-google; python_version >= "3.6"
certbot-dns-linode @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-linode; python_version >= "3.6"
certbot-dns-luadns @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-luadns; python_version >= "3.6"
certbot-dns-nsone @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-nsone; python_version >= "3.6"
certbot-dns-ovh @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-ovh; python_version >= "3.6"
certbot-dns-rfc2136 @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-rfc2136; python_version >= "3.6"
certbot-dns-route53 @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-route53; python_version >= "3.6"
certbot-dns-sakuracloud @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-dns-sakuracloud; python_version >= "3.6"
certbot-nginx @ file:///Users/bmw/Development/github.com/certbot/certbot/certbot-nginx; python_version >= "3.6"
```
I think I originally wrote the exact `acme` line because I could but I needed something a little more generic to match all the `certbot` lines.

Reply:
Cool thanks for explaining, lgtm.